rfcs/0005-artifact-revision-and-digest/README.md

```diff
@@ -0,0 +1,352 @@
+Must be one of `provisional`, `implementable`, `implemented`, `deferred`, `rejected`, `withdrawn`, or `replaced`.
+-->
+
+**Creation date:** 2022-10-20
+
+**Last update:** 2022-10-24
+
+## Summary
+
+This RFC proposes to establish a canonical format for an `Artifact` which
+points to a specific checksum (e.g. an OCI manifest digest or Git commit SHA)
+of a named pointer (e.g. an OCI image tag or Git tag). In addition, it proposes
+to include the algorithm name (e.g. `sha256`) as a prefix to any advertised
+checksum in an `Artifact` and further referring to it as a `Digest` as opposed
+to a `Checksum`.
+
+## Motivation
+
+The current `Artifact` type's `Revision` field format is not "officially"
+standardized (albeit assumed throughout our code bases), and has mostly been
+derived from `GitRepository` which uses `/` as a delimiter between the named
+pointer (a Git branch or tag) and a specific (SHA-1, or theoretical SHA-256)
+revision.
+
+Since the introduction of `OCIRepository` and with the recent changes around
+`HelmChart` objects to allow the consumption of charts from OCI registries.
+This could be seen as outdated or confusing due to the format differing from
+the canonical format used by OCI, which is `<tag>@<algo>:<checksum>` (the
+part after `@` formally known as a "digest") to refer to a specific version
+of a tagged OCI manifest.
+
+While also taking note that Git does not have an official canonical format for
+e.g. branch references at a specific commit, and `/` has less of a symbolic
+meaning than `@`. Which could be interpreted as "`<branch>` _at_
+`<commit SHA>`".
+
+In addition, with the introduction of algorithm prefixes for an `Artifact`'s
+checksum. It would be possible to add support and allow user configuration of
+other algorithms than SHA-256. For example SHA-384 and SHA-512, or the more
+performant (parallelizable) [BLAKE3][].
+
+Besides this, it would make it easier to implement a client that can verify the
+checksum without having to resort to an assumed format or guessing
+method based on the length of it, and allows for a more robust solution in
+which it can continue to calculate against the algorithm of a previous
+configuration.
+
+The inclusion of the `Artifact`'s algorithm prefix has been proposed before in
+[source-controller#855](https://github.com/fluxcd/source-controller/issues/855),
+with supportive response from Core Maintainers.
+
+### Goals
+
+- Establish a canonical format to refer to an `Artifact`'s `Revision` field
+  which consists of a named pointer and a specific checksum reference.
+- Allow easier verification of the `Artifact`'s checksum by including an
+  alias for the algorithm.
+- Allow configuration of the algorithm used to calculate the checksum of an
+  `Artifact`.
+- Allow configuration of [BLAKE3][] as an alternative to SHA for calculating
+  checksums. This has promising performance improvements over SHA-256, which
+  could allow for performance improvements in large scale environments.
```

Comment:
Why do we mix implementation details (i.e. a specific hash function) into this RFC?